src/qt3utils/datagenerators/spectrometers/princeston.py

```diff
@@ -228,100 +213,111 @@ def experiment_name(self, a_name: str) -> None:
+        self.light.set(lf.AddIns.CameraSettings.SensorTemperatureSetPoint, Double(deg_celsius))
 
     @property
     def exposure_time(self) -> float:
-        """
-        Returns the single frame exposure time (in ms).
-        """
         return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
 
     @exposure_time.setter
     def exposure_time(self, ms: float) -> None:
-        """
-        Sets the single frame exposure time to be ms (in ms).
-        """
         self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, Double(ms))
 
     @property
-    def sensor_temperature_set_point(self) -> float:
+    def num_frames(self) -> int:
         """
-        Returns the sensor setpoint temperature (Celsius).
+        Returns the number of frames taken during the acquisition.
         """
-        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+        return self.light.get(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore)
 
-    @sensor_temperature_set_point.setter
-    def sensor_temperature_set_point(self, deg_C: float) -> None:
+    @num_frames.setter
+    def num_frames(self, num_frames: int) -> None:
         """
-        Sets the sensor target temperature (in degrees Celsius) to deg_C.
-        This function retrieves image data, with particular attention to the camera's configuration determined by the `sensor_temperature_set_point`.
-        The `sensor_temperature_set_point` defines a target or reference value for the camera's sensor, ensuring optimal or specific operation conditions for image acquisition.
-        Depending on the setpoint, the behavior or response of the camera sensor might vary.
+        Sets the number of frames to be taken during acquisition to number.
         """
-        self.light.set(lf.AddIns.CameraSettings.SensorTemperatureSetPoint, Double(deg_C))
+        self.light.set(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore, Int64(num_frames))
 
-class SpectrometerDataAcquisition(SpectrometerConfig):
-    def acquire_frame(self) -> Tuple[np.ndarray, np.ndarray]:
-        """
-        Acquires a frame (or series of frames) from the spectrometer, and the
-        corresponding wavelength data.
-        """
+    def acquire(
+            self,
+            acquisition_mode: Literal['single', 'step-and-glue'],
+            wavelength_range: Tuple[float, float] = (0, 0),
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        return super().acquire(acquisition_mode, **{'wavelength_range': wavelength_range})
+
+    def single_acquisition(self) -> Tuple[np.ndarray, np.ndarray]:
         return self.light.acquire(), self.get_wavelengths()
-    
-    def acquire_step_and_glue(self, wavelength_range: Tuple[float, float]) -> Tuple[np.ndarray, np.ndarray]:
+
+    def step_and_glue_acquisition(
+            self,
+            wavelength_range: Tuple[float, float] = (0, 0)
+    ) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
We shouldn't make the default (0,0). If the user does not provide anything then it should raise and error

Reply:
The way the code is written in the new commit, it makes more sense to move this: 
```
self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueStartingWavelength, Double(lambda_min))
self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEndingWavelength, Double(lambda_max))
```
to the spectrometer config. The (0, 0) default was there prior to my refactoring. I think with if we define these in the config, there will be no need for defaults.

Reply:
I agree, we should move these. As for the (0,0) we remove it and we should be good on that front